server/lib/nl/detection/place.py

```diff
@@ -299,6 +312,23 @@ def get_place_from_dcids(place_dcids: List[str], debug_logs: Dict) -> any:
   return places, parent_places
 
 
+#
+# Helper function to get resolved entities that are not places.
+#
+def _get_non_place_entities(all_entities: List[str],
+                            resolved_places: List[Place]) -> List[Entity]:
+  entities = []
+  places = set([p.dcid for p in resolved_places])
+  non_place_entities = [e for e in all_entities if not e in places]
+  if non_place_entities:
+    names = property_values(non_place_entities, 'name')
+    for e in non_place_entities:
+      eNames = names.get(e, [])
+      eName = eNames[0] if len(eNames) > 0 else e
+      entities.append(Entity(dcid=e, name=eName, type=''))
```

Comment:
Why is there no type, can we get the type from the API so we can make downstream decisions?

Add TODO?

Reply:
since we currently weren't using the type, I wanted to avoid the extra API call, but added TODO

Reply:
Hmmm, I thought it doesn't need an extra API call --  the `property_values` I think doesn't expose the type, but the API call returns it? https://api.datacommons.org/v2/node?key=AIzaSyCTI4Xz-UW_G2Q2RfknhcfdAnTHq5X5XuI&nodes=geoId/06&property=-%3EcontainedInPlace

Reply:
but that's just for the containedInPlace property, name doesn't give the type: https://api.datacommons.org/v2/node?key=AIzaSyCTI4Xz-UW_G2Q2RfknhcfdAnTHq5X5XuI&nodes=geoId/06&property=-%3Ename

Reply:
Yes, `name` won't because its an `object_value`.   `object_id` things should return type, even beyond `containedInPlace`.  For example: https://api.datacommons.org/v2/node?key=AIzaSyCTI4Xz-UW_G2Q2RfknhcfdAnTHq5X5XuI&nodes=geoId/06&property=-%3EadministrativeCapital

Reply:
And we only want types for such IDs.

Reply:
I'm confused, we have a list of ids that we want to get types for, why would it not be an extra API call?

Reply:
Ohh I see.  I guess then we perhaps want a single Triple out call. But for later PR...